app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+        $this->line($process->getOutput());
+
+        if ($this->replaceEnvVariables()) {
+            $this->info('I replaced .env variables in your new Laravel application');
+        }
+
+        if ($this->option('auth')) {
+            $this->doAuth();
+        }
+
+        if ($this->option('createdb')) {
+            if (! in_array($this->option('createdb'), $this->dbtypes)) {
+                $this->info("You passed in '--createdb ".$this->option('createdb')."' but I do not understand");
+                $type = $this->anticipate('What type of database would you like to install? Options are '.implode(' or ', $this->dbtypes).'.', $this->dbtypes, $this->dbtypes[0]);
+            } else {
+                $type = $this->option('createdb');
+            }
+
+            if (! in_array($type, $this->dbtypes)) {
+                $this->alert("Now you're being silly. Entering $type, really? Okay, I won't create a database for you.");
+            } else {
+                $this->info("I am creating a new $type database");
+                if ($this->createDatabase($type)) {
+                    $this->info("I am executing 'php artisan migrate:fresh'");
+                    $this->migrateFresh();
+                }
+            }
+        }
+
+        if ($this->option('node')) {
+            $this->doNodeOrYarn();
+        }
+
+        if ($this->option('link')) {
+            if ($this->tools['valet']) {
+                if ($this->doValetLink()) {
+                    $this->openBrowser();
+                }
+            }
+        }
+
+        if ($this->tools['git']) {
+            $this->doGit();
+        }
+
+        $this->openTextEditor();
+
+        $this->info("You're ready to go! Remember to cd into '{$this->projectpath}' before you start editing.");
+    }
+
+    protected function getAvailableTools()
+    {
+        $finder = new ExecutableFinder();
+
+        $checks = ['yarn', 'npm', 'git', 'valet', 'laravel', 'composer'];
+        foreach ($checks as $check) {
+            $this->tools[$check] = $finder->find($check);
+        }
+    }
+
+    protected function setBasePath()
+    {
+        $this->basepath = $this->cwd;
+        if ($this->option('path')) {
+            $path = $this->option('path');
+            if (is_dir($path)) {
+                $this->basepath = $path;
+            } else {
+                $this->warn("Your defined '--path $path' is not a directory, so I am skipping it and using '{$this->basepath}' instead.");
+            }
+        }
+
+        return $this->basepath;
+    }
+
+    protected function askToAndRemoveProject()
+    {
+        $this->info("The directory '{$this->projectpath}' already exists.");
+
+        if ($this->confirm("Shall I proceed by removing the following directory? {$this->projectpath}")) {
+            $fs = new Filesystem\Filesystem();
+
+            // @todo Use laravel --force here instead of deleteDirectory()?
```

Comment:
OO didn't know that exists! Definitely.

Reply:
I propose we leave this topic for later when the laravel installer becomes optional so folks can use composer instead. Okay?

Reply:
:) Sounds fine!